### PR TITLE
Update ShadowTranslateBehavior.php

### PR DESCRIPTION
**Pull request by greew**
Update ```ShadowTranslateBehavior::groupTranslations()``` to correctly mimic the updated TranslateBehaviorTest and make sure it doesn't crash on the newly added test in the TranslateBehaviorTest.

### CODE REVIEW

File: src/Model/Behavior/ShadowTranslateBehavior.php [hunk 1]
```diff
@@ -378,6 +378,9 @@ public function groupTranslations($results)
     {
         return $results->map(function ($row) {
             $translations = (array)$row['_i18n'];
+            if (empty($translations) && $row->get('_translations')) {
```
Comment (dereuromark) [inline src/Model/Behavior/ShadowTranslateBehavior.php]: After a recent discussion about potentially hidden bugs using empty, it is usually encouraged to use sth else.
E.g.

    if (!$translations && ...)

or

 if (count($translations) === 0 && ...)

Comment (greew) [inline src/Model/Behavior/ShadowTranslateBehavior.php]: I'll happily change it - I just copied it directly from https://github.com/cakephp/cakephp/blob/master/src/ORM/Behavior/TranslateBehavior.php#L460 :)

Comment (greew) [inline src/Model/Behavior/ShadowTranslateBehavior.php]: And btw... the above mentioned discussion.. can I see that somewhere online?

Comment (dereuromark) [inline src/Model/Behavior/ShadowTranslateBehavior.php]: Yes, of course: https://github.com/cakephp/cakephp/issues/8397

**Comment by greew** [comment]
Ping @ad7six

**Pull request merged**